### PR TITLE
Fixes #2198: Updated README page and restructured documentation sections

### LINKED ISSUE #2198: Improve README page
Improve README page, for example:
* Picture:
  * Remove wbemcli
  * Add mock support
  * Use consistent names
  * Add a WBEM server at the bottom to avoid confusion about WBEM Server box
 * Update section below the picture to match updated picture
* Quick start section:
  * Rename server_uri to server_url
* Documentation section:
  * Remove link to latest unreleased dev docs - too complex to distinguish
  * Put docs, change log, tutorial, presentation links in a simple list
* Command line tools section:
  * Add a link to pywbemcli

### PR DESCRIPTION
**Pull request by andy-maier**
See commit message.

### CODE REVIEW

File: docs/intro.rst [hunk 1]
```diff
@@ -13,40 +13,65 @@ Introduction
...
+* :ref:`WBEM client library`
+
+  An API that supports issuing WBEM operations to a WBEM server, using the CIM
   operations over HTTP (CIM-XML) protocol defined in the DMTF standards
   :term:`DSP0200` and :term:`DSP0201`.
 
-* :ref:`WBEM server API`
+* :ref:`WBEM server library`
+
+  An API that encapsulates certain functionality of a WBEM server for use by a
```
Comment (KSchopmeyer) [inline docs/intro.rst:30]: Change certain to selected

File: pywbem/_server.py [hunk 2]
```diff
@@ -17,17 +17,20 @@
 """
 *New in pywbem 0.9 as experimental and finalized in 0.10.*
 
-The WBEM server API encapsulates certain functionality of a WBEM server for use
-by a WBEM client application, such as determining the Interop namespace of the
-server, or the management profiles advertised by the server.
+The WBEM server library API of pywbem encapsulates certain functionality of a
```
Comment (KSchopmeyer) [inline pywbem/_server.py:20]: Certain to selected

Comment (andy-maier) [reply pywbem/_server.py:20]: DONE

File: docs/intro.rst (hunk 1 above)
Comment (andy-maier) [reply docs/intro.rst:30]: DONE

**Comment by andy-maier** [comment]
Resolved the comments and another occurrence in README.rst in the latest update.

**Pull request merged**